### PR TITLE
Added doc for cmor and fix preprocessor steps

### PR DESCRIPTION
**Pull request by jvegasbsc**
Just progessing a bit in #193

### CODE REVIEW

File: doc/esmvalcore/preprocessor.rst [hunk 1]
```diff
@@ -111,9 +111,55 @@ See also :func:`esmvalcore.preprocessor.derive`.
...
 ======================================
-.. warning::
 
-   Section to be added.
+Data checking
+-------------
+
+Data preprocessed by ESMValTool is automatically checked against its cmor
+definition. To reduce the impact of this check while maintaing it as realiable
+as possible, it is splitted in two parts: one will check the metadata and will
```
Comment (bouweandela) [inline doc/esmvalcore/preprocessor.rst]:
```suggestion
as possible, it is split in two parts: one will check the metadata and will
```

File: doc/esmvalcore/preprocessor.rst [hunk 2]
```diff
@@ -111,9 +111,55 @@ See also :func:`esmvalcore.preprocessor.derive`.
...
+Checks include, but are not limited to:
+
+   - Requested coordinates are present and comply with their definition.
+   - Correctness of variable names, units and other metadata.
+   - Compliance with the valid minimum and maximum values allowed if defined.
+
+The most relevant (i.e. a missing coordinate) will raise an error while others
+(i.e an incorrect long name) will be reported as a warning.
+
+Somo of those issues will be fixed automatically by the tool, including the
```
Comment (bouweandela) [inline doc/esmvalcore/preprocessor.rst]:
```suggestion
Some of those issues will be fixed automatically by the tool, including the
```

File: doc/esmvalcore/preprocessor.rst [hunk 3]
```diff
@@ -111,9 +111,55 @@ See also :func:`esmvalcore.preprocessor.derive`.
...
+    - Direction of coordinates.
+    - Automatic clipping of longitude to 0 - 360 interval.
+
+
+Dataset specific fixes
+----------------------
+
+Sometimes, the checker will detect errors that it can not fix by itself.
+ESMValTool deals with those issues by applying specific fixes for those
+dataset that require them. Fixes are applied at three different preprocessor
```
Comment (bouweandela) [inline doc/esmvalcore/preprocessor.rst]:
```suggestion
datasets that require them. Fixes are applied at three different preprocessor
```

File: doc/esmvalcore/preprocessor.rst [hunk 4]
```diff
@@ -111,9 +111,55 @@ See also :func:`esmvalcore.preprocessor.derive`.
...
+steps:
+
+    - fix_file: apply fixes directly to a copy of the file. Only fixes that allow
```
Comment (bouweandela) [inline doc/esmvalcore/preprocessor.rst]: change to something like: 'Only fixes that prevent iris from loading the file are applied here.'

File: doc/esmvalcore/preprocessor.rst [hunk 5]
```diff
@@ -111,9 +111,55 @@ See also :func:`esmvalcore.preprocessor.derive`.
...
+      iris to load the file are applied here.
```
Comment (bouweandela) [inline doc/esmvalcore/preprocessor.rst]: Add a link to https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.preprocessor.html#esmvalcore.preprocessor.fix_file

File: doc/esmvalcore/preprocessor.rst [hunk 6]
```diff
@@ -111,9 +111,55 @@ See also :func:`esmvalcore.preprocessor.derive`.
...
+
+    - fix_metadata: metadata fixes are done just before concatenating the cubes
```
Comment (bouweandela) [inline doc/esmvalcore/preprocessor.rst]: Add link to preprocessor function

File: doc/esmvalcore/preprocessor.rst [hunk 7]
```diff
@@ -111,9 +111,55 @@ See also :func:`esmvalcore.preprocessor.derive`.
...
+      loaded from different files in the final one. Automatic metadata fixes
+      are also applied at this step.
+
+    - fix_data: data fixes are applied just prior to checking the data.
```
Comment (bouweandela) [inline doc/esmvalcore/preprocessor.rst]: Add link to preprocessor function

Comment (bouweandela) [reply doc/esmvalcore/preprocessor.rst]: Also, data fixes are applied just after time selection, while the check is done very late in the preprocessing, see https://github.com/ESMValGroup/ESMValCore/blob/9f1c1e489496328dfb97d82a58d34a9660cb1b1a/esmvalcore/preprocessor/__init__.py#L49-L85

**Comment by jvegasbsc** [comment]
Done!

**Pull request merged**